metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/tree_function.py

```diff
@@ -70,50 +70,30 @@ def internal(node: Union[Tree, Token]) -> Optional[Tree]:
     return expression_tree
 
 
-def token_values(tree: Tree, parameters: Optional[Dict[str, str]] = None) -> List[str]:
+def token_values(tree: Tree, parameters: Dict[str, str] = {}) -> List[str]:
     """
 
     :param tree: Tree to traverse
     :param parameters: If parameters is not None, it will try to resolve identifier variable references
                        using the values in 'parameters'.
+                       Read more about parameters feature of PowerBI M-Query at https://learn.microsoft.com/en-us/power-query/power-query-query-parameters
     :return: List of leaf token data
     """
     values: List[str] = []
 
     def internal(node: Union[Tree, Token]) -> None:
-        if (
-            parameters is not None
-            and isinstance(node, Tree)
-            and node.data == "identifier"
-            and node.children[0].data == "quoted_identifier"
-        ):
-            # This is the case where they reference a variable using
-            # the `#"Name of variable"` syntax.
-
-            identifier = node.children[0].children[0]
-            assert isinstance(identifier, Token)
-
-            # For quoted_identifier, ref will have quotes around it.
-            # However, we'll probably need to expand this to all identifier types,
-            # which are not required to have quotes (using make_function_name).
-            ref = identifier.value
-            if ref.startswith('"') and ref[1:-1] in parameters:
-                resolved = parameters[ref[1:-1]]
-                values.append(resolved)
-            elif ref in parameters:
-                resolved = parameters[ref]
-                values.append(resolved)
-            else:
-                # If we can't resolve, fall back to the name of the variable.
-                logger.debug(f"Unable to resolve parameter reference to {ref}")
-                values.append(ref)
-        elif isinstance(node, Token):
-            # This means we're probably looking at a literal.
-            values.append(cast(Token, node).value)
+        if isinstance(node, Token):
+            node_value: Any = cast(Token, node).value
+            # Check if the node_value is actually a parameter in parameters dict
+            # if present then resolve it to parameter value
+            parameter_key: str = node_value.strip('"')
+            if parameter_key in parameters:
+                node_value = parameters[parameter_key]
```

Comment:
this code conflates normal strings e.g. `"my-bq-project"` with variable references `#"Parameter - BQ Proj ID"`, which the previous code did not

I think it probably makes sense to revert this to the previous version

Reply:
We always get identifier without # , so either identifier is my-bp-project or "Parameter - BQ Proj ID". I verified it by creating parameter with and without space in PowerBI desktop and also test-case that you have added and new test-case that I have added are working fine.

Reply:
It can be `MyParameter` or `#"MyParameter"`. However, it is still true that `"my-bq-project"` is still a raw string and this could produce incorrect results

Reply:
this should retain the check for .data == "identifier" - it should be identifier instead of quoted_identifier

Reply:
done